solarwinds/stackstate_checks/solarwinds/solarwinds.py

```diff
@@ -0,0 +1,486 @@
+                dest_interface_id = npm_node.get("DestInterfaceID")
+                if dest_interface_id:
+                    # Is the destination node (of this interface) present in the solarwinds data?
+                    found_dest_node = next((sub for sub in npm_data if sub["InterfaceID"] == dest_interface_id), None)
+                    if found_dest_node:
+                        dest_node_name = found_dest_node.get("Caption")
+                        dest_interface_name = found_dest_node.get("IfName")
+                        dest_mac_address = npm_node.get("DestInterfaceMAC")
+                        found_interface.append_connection(dest_node_name, dest_interface_name, dest_mac_address)
+        return nodes
+
+    def check_udt_active(self):
+        self.log.info("Check for installed UDT module")
+        # Get a list of all installed Orion modules, check for (not expired) UDT
+        query_udt = self.swis.query(
+            "SELECT LicenseName, IsExpired FROM Orion.InstalledModule WHERE LicenseName = 'UDT' AND IsExpired = FALSE"
+        )
+        if query_udt["results"]:
+            self.log.info("UDT module is active")
+            return True
+        else:
+            self.log.info("Licensed UDT module not detected")
+            return False
+
+    def get_udt_topology_data(self):
+        self.log.info("Collect UDT connection information")
+        # Get unique connections from UDT
+        # This shows nodes connected to Switch ports (that may not be monitored inside Orion)
+        query_udt_connections = self.swis.query(
+            "SELECT DISTINCT ConnectedTo, ConnectionTypeName, HostName, IPAddress, MACAddress, PortNumber "
+            "FROM Orion.UDT.AllEndpoints WHERE IPAddress IS NOT NULL"
+        )
+        # Assign the query results to a variable
+        udt_connections = query_udt_connections["results"]
+        return udt_connections
+
+    def add_udt_topology(self, udt_topology_data, npm_topology):
+        self.log.info("Add UDT topology information to NPM topology data")
+        for udt_connection in udt_topology_data:
+            src_node_name = OrionComponent.strip_domain(udt_connection.get("ConnectedTo"))
+            src_interface_name = udt_connection.get("PortNumber")
+            dest_mac_address = udt_connection.get("MACAddress").replace(":", "")
+            dest_ip_address = udt_connection.get("IPAddress")
+            dest_node_name = udt_connection.get("HostName")
+            if not dest_node_name:
+                dest_node_name = dest_ip_address
+
+            # Find the source node to connect from
+            found_src_node = next((sub for sub in npm_topology if sub.caption == src_node_name), None)
+            if found_src_node:
+                # Find the interface to connect from
+                found_src_interface = next(
+                    (sub for sub in found_src_node.interfaces if sub.interface_name == src_interface_name), None
```

Comment:
same here.

Reply:
check comment above about generator and next function